### PR TITLE
Cascade down dates from document

### LINKED ISSUE #49: Land document dates to statements on merge
When merging documents, any statements without any dates should inherit the document date. Currently, statements which don't have timestamps will be merged without dates, corrupting the VEX history.

Here is a sample merged doc:

```json
{
  "id": "vex-8a489e826e47a7a9bc8c4a7b37a2d8033360f06c29f65433d17794a881e2d735",
  "format": "text/vex",
  "author": "",
  "role": "",
  "timestamp": "2022-12-28T13:31:27.77315253-06:00",
  "statements": [
    {
      "vulnerability": "CVE-1234-5678",
      "products": [
        "pkg:apk/wolfi/git@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-daemon@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-email@2.39.0-r0?distro=wolfi"
      ],
      "status": "not_affected"
    }
}

```

### PR DESCRIPTION
**Pull request by puerco**
This PR changes the merge code to cascade dates down from the document when combined statements don't have one.

/cc @luhring 
Fixes #49

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

### CODE REVIEW

File: pkg/ctl/implementation.go [hunk 1]
```diff
@@ -355,6 +357,13 @@ func (impl *defaultVexCtlImplementation) Merge(
 				}
 			}
 
+			// If statement does not have a timestamp, cascade
+			// the timestamp down from the document.
+			// See https://github.com/chainguard-dev/vex/issues/49
+			if s.Timestamp == nil {
+				s.Timestamp = doc.Timestamp
+			}
```
Comment (luhring) [inline pkg/ctl/implementation.go:365]: Question: It's not a guarantee in this part of the code that `doc.Timestamp` is not `nil`, is that right? I'm thinking we might want to require a non-nil `doc.Timestamp` before we apply it to the statement. If the document doesn't have a timestamp, and the statement doesn't have a timestamp, we know we'll be producing a corrupt VEX history, and should probably error out. WDYT?

Comment (puerco) [reply pkg/ctl/implementation.go]: Sounds good to me! 
I've added a check and error before assigning it blindly

**Review by luhring — APPROVED** [review]
Nice! 👏

File: pkg/ctl/implementation.go (hunk 1 above)
Comment (luhring) [reply pkg/ctl/implementation.go]: Thanks!

**Pull request merged**